python/tank/context.py

```diff
@@ -984,7 +1061,7 @@ def context_yaml_constructor(loader, node):
 ################################################################################################
 # utility methods
 
-def _task_from_sg(tk, task_id):
+def _task_from_sg(tk, task_id, additional_fields = None):
```

Comment:
I think if it is an optimisation then keep it the way it is - maybe add a comment mentioning that!